packages/graphql-typescript-definitions/src/index.ts

```diff
@@ -189,6 +164,80 @@ export class Builder extends EventEmitter {
     await this.generateDocumentTypes();
   }
 
+  stop() {
+    this.watchers.forEach((watcher) => {
+      watcher.close();
+    });
+
+    this.watchers.length = 0;
+  }
+
+  private setupDocumentWatchers() {
+    const updateForProject = async (
+      filePath: string,
+      projectConfig: GraphQLProjectConfig,
+    ) => {
+      try {
+        await this.updateDocumentForFile(filePath, projectConfig);
+      } catch (error) {
+        this.emit('error', error);
+        return;
+      }
+
+      await this.generateDocumentTypes();
+    };
+
+    return getGraphQLProjects(this.config).map((projectConfig) => {
+      function update(filePath: string) {
+        return updateForProject(filePath, projectConfig);
+      }
+
+      return watch(
+        projectConfig.includes.map((include) =>
+          getGraphQLFilePath(this.config, include),
+        ),
+        {
+          ignored: projectConfig.excludes.map((exclude) =>
+            getGraphQLFilePath(this.config, exclude),
+          ),
+          ignoreInitial: false,
+        },
+      )
+        .on('add', update) // TODO: async function not supported
+        .on('change', update) // TODO: async function not supported
```

Comment:
this here is problematic because `chokidar` does not support async event handling. so we are leaking promises like crazy on `change` and `add` events.